### PR TITLE
Upgrade @guardian/types to use Theme

### PR DESCRIPTION
**Pull request by webb04**
## Why are you doing this?
- Updates `@guardian/types` to `0.5.2`
- Rename pillar helper functions to `theme`
- Removed `featureHeadline` and `soft` colours that aren't being used
- This should let us start work on special reports

### CODE REVIEW

**Review by frankie297 — APPROVED** [review]
Do we have somewhere where we can add a definition of what is the difference between `Theme` and `Pillar`?

**Comment by webb04** [comment]
> Do we have somewhere where we can add a definition of what is the difference between `Theme` and `Pillar`?

Not sure if we have any documentation around this @JamieB-gu ?

The types repo defines it here:
https://github.com/guardian/types/blob/main/Format.ts#L3-L15

We have the problem where the special report colours need to override pillar colours on special report articles.

It might be confusing to call them a `pillar` as we don't treat them as pillars in some places like CAPI and the navigation bar on dotcom for example. So we updated `Format` to use a `Theme` which can be a `Pillar` or something `Special`

File: src/components/media/byline.tsx [hunk 1]
```diff
@@ -39,13 +38,12 @@ const styles: SerializedStyles = css`
 // ----- Component ----- //
 
 interface Props {
-    pillar: Pillar;
     publicationDate: Option<Date>;
     className: SerializedStyles;
     item: Item;
 }
 
-const Byline: FC<Props> = ({ pillar, publicationDate, className, item }) => {
```
Comment (JamieB-gu) [inline src/components/media/byline.tsx:48]: Interesting that this didn't previously throw an unused argument warning 🤔

File: src/pillarStyles.ts [hunk 2]
```diff
@@ -60,10 +60,18 @@ export const pillarColours: PillarColours = {
...
+        liveblogBackground: palette.lifestyle[300],
+        liveblogDarkBackground: palette.lifestyle[200]
     }
 }
 
-const getPillarStyles = (pillar: Pillar): PillarStyles => pillarColours[pillar];
+const getPillarStyles = (pillar: Theme): PillarStyles => pillarColours[pillar];
 
 function pillarFromString(pillar: string | undefined): Pillar {
     switch (pillar) {
```
Comment (JamieB-gu) [inline src/pillarStyles.ts:77]: Some of the names in this file might be a bit confusing now? For example, `PillarColours` where the key is `theme`?

**Review by JamieB-gu — APPROVED** [review]
Herculean refactor here, 💪

**Comment by JamieB-gu** [comment]
> Do we have somewhere where we can add a definition of what is the difference between `Theme` and `Pillar`?

> Not sure if we have any documentation around this

It's a good point. We should probably put some markdown docs into the `@guardian/types` repo.

**Comment by JamieB-gu** [comment]
@webb04 you may find the conversation [here](https://github.com/guardian/types/issues/4) useful for future work on this.

**Pull request merged**